plugins/outputs/azuremonitor/azuremonitor.go

```diff
@@ -0,0 +1,490 @@
+							newAzureMonitorSeries(dimensionValues, fv),
+						},
+					},
+				},
+			}
+		} else {
+			tmp, i, ok := azm.findSeries(dimensionValues)
+			if !ok {
+				// add series new series (should be rare)
+				n := append(azm.Data.BaseData.Series, newAzureMonitorSeries(dimensionValues, fv))
+				a.cache[name].Data.BaseData.Series = n
+				continue
+			}
+
+			//counter compute
+			n := tmp.Count + 1
+			tmp.Count = n
+			//max/min compute
+			if fv < tmp.Min {
+				tmp.Min = fv
+			} else if fv > tmp.Max {
+				tmp.Max = fv
+			}
+			//sum compute
+			tmp.Sum += fv
+			//store final data
+			a.cache[name].Data.BaseData.Series[i] = tmp
+		}
+	}
+}
+
+func (m *azureMonitorMetric) findSeries(dv []string) (*azureMonitorSeries, int, bool) {
+	if len(m.Data.BaseData.DimensionNames) != len(dv) {
+		return nil, 0, false
+	}
+	for i := range m.Data.BaseData.Series {
+		if m.Data.BaseData.Series[i].equal(dv) {
+			return m.Data.BaseData.Series[i], i, true
+		}
+	}
+	return nil, 0, false
+}
+
+func newAzureMonitorSeries(dv []string, fv float64) *azureMonitorSeries {
+	return &azureMonitorSeries{
+		DimensionValues: append([]string{}, dv...),
+		Min:             fv,
+		Max:             fv,
+		Sum:             fv,
+		Count:           1,
+	}
+}
+
+func (s *azureMonitorSeries) equal(dv []string) bool {
+	if len(s.DimensionValues) != len(dv) {
+		return false
+	}
+	for i := range dv {
+		if dv[i] != s.DimensionValues[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func convert(in interface{}) (float64, bool) {
+	switch v := in.(type) {
+	case int:
+		return float64(v), true
+	case int8:
+		return float64(v), true
+	case int16:
+		return float64(v), true
+	case int32:
+		return float64(v), true
+	case int64:
+		return float64(v), true
+	case uint:
+		return float64(v), true
+	case uint8:
+		return float64(v), true
+	case uint16:
+		return float64(v), true
+	case uint32:
+		return float64(v), true
+	case uint64:
+		return float64(v), true
+	case float32:
+		return float64(v), true
+	case float64:
+		return v, true
+	case string:
+		f, err := strconv.ParseFloat(v, 64)
+		if err != nil {
+			log.Printf("converted string: %s to %v", v, f)
+			return 0, false
+		}
+		return f, true
+	default:
+		log.Printf("did not convert %T: %s", v, v)
+		return 0, false
+	}
+}
+
+func (a *AzureMonitor) push() {
+	var body []byte
+	for _, metric := range a.cache {
+		jsonBytes, err := json.Marshal(&metric)
+		if err != nil {
+			log.Printf("Error marshalling metrics %s", err)
+			return
+		}
+		body = append(body, jsonBytes...)
+		body = append(body, '\n')
+	}
+
+	_, err := a.postData(&body)
+	if err != nil {
+		log.Printf("Error publishing aggregate metrics %s", err)
+	}
+	return
+}
+
+func (a *AzureMonitor) postData(msg *[]byte) (*http.Request, error) {
+	if err := a.validateCredentials(); err != nil {
+		return nil, fmt.Errorf("Error authenticating: %v", err)
+	}
+
+	metricsEndpoint := fmt.Sprintf("https://%s.monitoring.azure.com%s/metrics",
```

Comment:
One thing to be careful here -- As custom metrics is in preview on the Azure Monitor side, we don't support all regions of public Azure.  We are only going to be available for a few regions as part of the reviews so not all these endpoints will exist.